pkg/client/observatorium/observability_config.go

```diff
@@ -94,6 +151,46 @@ func (c *ObservabilityCloudWatchLoggingConfig) validate() error {
 	return nil
 }
 
+func (c *DataPlaneObservabilityConfig) validate() error {
+	if !c.Enabled {
+		return nil
+	}
+
+	validate := validator.New()
+
+	oidcConfig := c.RemoteWriteConfiguration.Authentication.OIDC
+	if oidcConfig != nil && oidcConfig.Static != nil && oidcConfig.Autogenerated != nil {
```

Comment:
I intended to use a custom type validator, but couldn't make it work with Go playground validator and the docs for it are lacking.

Reply:
This can be achieved by using a struct level validator with the `RegisterStructValidation` method.
See an example in https://github.com/go-playground/validator/blob/master/_examples/struct-level/main.go#L80